src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableHttpClientCRLCheck.cs

```diff
@@ -0,0 +1,218 @@
+            ImmutableArray.Create(
+                DefinitelyDisableHttpClientCRLCheckRule,
+                MaybeDisableHttpClientCRLCheckRule);
+
+        /// <summary>
+        /// PropertySetAbstractValue index for CheckCertificateRevocationList property.
+        /// </summary>
+        private const int CheckCertificateRevocationListIndex = 0;
+
+        private static readonly ConstructorMapper ConstructorMapper = new ConstructorMapper(
+            (IMethodSymbol constructorMethod, IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+            {
+                return PropertySetAbstractValue.GetInstance(PropertySetAbstractValueKind.Flagged);
+            });
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "CheckCertificateRevocationList",
+                (ValueContentAbstractValue valueContentAbstractValue) =>
+                    PropertySetCallbacks.EvaluateLiteralValues(
+                        valueContentAbstractValue,
+                        (object o) => o is bool booleanValue && booleanValue == false),
+                CheckCertificateRevocationListIndex));
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+            new HazardousUsageEvaluator(
+                WellKnownTypeNames.SystemNetHttpHttpClient,
+                ".ctor",
+                "handler",
+                PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown));
+
+        private static readonly ImmutableHashSet<string> typeToTrackMetadataNames = ImmutableHashSet.Create<string>(
+            WellKnownTypeNames.SystemNetHttpWinHttpHandler,
+            WellKnownTypeNames.SystemNetHttpHttpClientHandler,
+            WellKnownTypeNames.SystemNetHttpUnixCurlHandler);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(WellKnownTypeNames.SystemNetHttpHttpClient, out INamedTypeSymbol httpClientTypeSymbol))
+                    {
+                        return;
+                    }
+
+                    if (typeToTrackMetadataNames.Any(s => !wellKnownTypeProvider.TryGetTypeByMetadataName(s, out _)))
+                    {
+                        return;
+                    }
+
```

Comment:
Given the handlers are (maybe?) in different assemblies, I don't think it makes sense to skip analysis if not all of them are available.